dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -0,0 +1,180 @@
+      String fileName) throws IOException {
+    ClassParser parser = new ClassParser(classFileStream, fileName);
+    JavaClass javaClass = parser.parse();
+    return listConstantPoolMethodRef(javaClass);
+  }
+
+  /**
+   *  Lists the methodref entries owned by the defined class in class file
+   *
+   * @param classFileStream stream of a class file
+   * @param fileName name of the file that contains class
+   * @return methodref entries owned by the class
+   * @throws IOException when there is problem in reading classFileStream
+   */
+  public static List<ConstantPoolMethodref> listOwningConstantPoolMethodref(InputStream classFileStream,
+      String fileName) throws IOException {
+    ClassParser parser = new ClassParser(classFileStream, fileName);
+    JavaClass javaClass = parser.parse();
+    List<ConstantPoolMethodref> methodrefs = listConstantPoolMethodRef(javaClass);
+    Set<String> owningClass = Sets.newHashSet(javaClass.getClassName());
+    List<ConstantPoolMethodref> owningMethodrefs =  methodrefs
+            .stream()
+            .filter(methodref -> owningClass.contains(methodref.getClassName()))
+            .collect(Collectors.toList());
+    return owningMethodrefs;
+  }
+
+  /**
+   *  Lists the methodref entries not owned by the defined class in class file
+   *  The list the class attemts to invoke the methods defined elsewhere
+   *
+   * @param classFileStream stream of a class file
+   * @param fileName name of the file that contains class
+   * @return methodref entries external to the class
+   * @throws IOException when there is problem in reading classFileStream
+   */
+  public static List<ConstantPoolMethodref> listExternalConstantPoolMethodref(
+      InputStream classFileStream, String fileName) throws IOException {
+    ClassParser parser = new ClassParser(classFileStream, fileName);
+    JavaClass javaClass = parser.parse();
+    List<ConstantPoolMethodref> methodrefs = listConstantPoolMethodRef(javaClass);
+    Set<String> owningClass = Sets.newHashSet(javaClass.getClassName());
+    List<ConstantPoolMethodref> owningMethodrefs =
+        methodrefs
+            .stream()
+            .filter(methodref -> !owningClass.contains(methodref.getClassName()))
+            .collect(Collectors.toList());
+    return owningMethodrefs;
+  }
+
+  /**
+   * Lists methods entries defined in the class file
+   *
+   * @param classFileStream stream of a class file
+   * @param fileName name of the file that contains class
+   * @return method and signature entries defined in the class file
+   * @throws IOException when there is problem in reading classFileStream
+   */
+  public static List<MethodAndSignature> listDeclaredMethods(InputStream classFileStream, String fileName)
+      throws IOException {
+    final ClassParser parser = new ClassParser(classFileStream, fileName);
+    final JavaClass javaClass = parser.parse();
+    Method[] methods = javaClass.getMethods();
```

Comment:
Does this return private methods in the result?

Reply:
I think we need to treat them specially. We could have a situation where a public method becomes private, in which case it was a breaking change. Providing an error message that it doesn't exist could potentially be confusing. This is distinct from a public method being renamed to something else - the symbol is completely gone. So, we might want to return it and track that it is private somehow. (I would suggest making this a later enhancement though; I was just curious at this point.)

Reply:
Sure, I'll make it later enhancement.